# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webscheduling/WebTaskSchedulerWorker.cpp
# Commit: 4739af17e757
# Full Hash: 4739af17e757724c0833a8564579833249389566
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2025-02-27 21:52:52
# Regressor Bug: 1920115
# File Overlap Count: 2
# Description:
#   Bug 1920115 - Allow high priority tasks to run before timers r=smaug
#   
#   In Gecko, unlike other runnables where the TaskController
#   picks one each at a time, the TimeoutManager allows multiple
#   timers to run sequentially without allowing other runnables
# ==============================================================================

diff -r 0773f669c498 -r 4739af17e757 dom/webscheduling/WebTaskSchedulerWorker.cpp
--- a/dom/webscheduling/WebTaskSchedulerWorker.cpp	Thu Feb 27 16:17:36 2025 +0000
+++ b/dom/webscheduling/WebTaskSchedulerWorker.cpp	Thu Feb 27 16:17:36 2025 +0000
@@ -9,7 +9,6 @@
 #include "mozilla/dom/TimeoutManager.h"
 
 namespace mozilla::dom {
-
 WebTaskWorkerRunnable::WebTaskWorkerRunnable(
     WebTaskSchedulerWorker* aSchedulerWorker)
     : WorkerSameThreadRunnable("WebTaskWorkerRunnable"),
@@ -113,4 +112,16 @@
   }
   WebTaskScheduler::Disconnect();
 }
+
+void WebTaskSchedulerWorker::
+    IncreaseNumNormalOrHighPriorityQueuesHaveTaskScheduled() {
+  ++mNumHighPriorityQueuesHaveTaskScheduled;
+}
+
+void WebTaskSchedulerWorker::
+    DecreaseNumNormalOrHighPriorityQueuesHaveTaskScheduled() {
+  MOZ_ASSERT(mNumHighPriorityQueuesHaveTaskScheduled > 0);
+  --mNumHighPriorityQueuesHaveTaskScheduled;
+}
+
 }  // namespace mozilla::dom